### PR TITLE
Update module release docs

### PR DESCRIPTION
**Pull request by bastelfreak**
* Now mentions gpg
* Properly explains our changelog generator

### CODE REVIEW

File: _docs/releasing_version.md [hunk 1]
```diff
@@ -49,6 +47,10 @@ the changelog generator expects the token in the environment variable `CHANGELOG
 CHANGELOG_GITHUB_TOKEN='mytoken' bundle exec rake changelog
 ```
 
+The changelog generator checks for certain labels on closed issues and PRs since the last release and groups them together. Backwards incompatible changes require a major release.
+If only bug fixes were merged, a patch release is suggested. If both isn't the case, continue with a minor release. Check the generated diff for the CHANGELOG.md. If your chosen
```
Comment (trevor-vaughan) [inline _docs/releasing_version.md:51]: Not quite following `If both isn't the case`. Both what?

File: _docs/releasing_version.md [hunk 2]
```diff
@@ -49,6 +47,10 @@ the changelog generator expects the token in the environment variable `CHANGELOG
...
+release version doesn't fit to the generated changelog, update metadata.json and run the changelog task again.
```
Comment (trevor-vaughan) [inline _docs/releasing_version.md:52]:
```suggestion
release version doesn't match the generated changelog, update metadata.json and run the changelog task again.
```

File: _docs/releasing_version.md (hunk 1 above)
Comment (bastelfreak) [reply _docs/releasing_version.md:51]: ah. What I meant: If the Changelog does not
* only contain merged PRs that are labeled `bug` 
* only contain merged PRs that are labeled `backwards-incompatible`

then go on with a minor release

Comment (bastelfreak) [reply _docs/releasing_version.md:51]: let me know if you've a better wording in mind :)

Comment (kenyon) [reply _docs/releasing_version.md:51]: > If the changes were neither backwards-incompatible nor only bug fixes, make a minor release.

Comment (bastelfreak) [reply _docs/releasing_version.md]: updated it

**Review by ekohl — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

**Pull request merged**